### PR TITLE
Rename indexes-of to offsets-of. Add tests for offsets-of. Relates to #117.

### PR DESCRIPTION
**Pull request by anthonygalea**

### CODE REVIEW

File: src/rethinkdb/query.cljc [hunk 1]
```diff
@@ -347,11 +347,11 @@
   [sq n]
   (term :NTH [sq n]))
 
-(defn indexes-of
+(defn offsets-of
```
Comment (danielcompton) [inline src/rethinkdb/query.cljc]: Can you add a fallback `def` for indexes-of and add `^:deprecated` metadata?

File: src/rethinkdb/query.cljc [hunk 2]
```diff
@@ -347,11 +347,11 @@
...
   "Get the indexes of an element in a sequence. If the argument is a predicate,
   get the indexes of all elements matching it."
   [sq obj-or-func]
-  (term :INDEXES_OF [sq obj-or-func]))
+  (term :OFFSETS_OF [sq obj-or-func]))
```
Comment (danielcompton) [inline src/rethinkdb/query.cljc]: Has the term changed too? Do we need to update the protobuf for this?

Comment (anthonygalea) [inline src/rethinkdb/query.cljc]: I don't see TermType INDEXES_OF in https://github.com/apa512/clj-rethinkdb/blob/master/src/rethinkdb/types.cljc

**Comment by danielcompton** [comment]
Looks like the protobuf was updated in May last year https://github.com/apa512/rethinkdb-protobuf/commit/ad92ff4bd621ae323f3b1ef056a6ff542c4c7b8c. A great reason why tests on all the functions is helpful :)

**Comment by anthonygalea** [comment]
👍

**Pull request merged**